byway/src/main/webapp/destinations.js

```diff
@@ -234,5 +241,10 @@ else{
   initializeDestinationsPage();
 }
 
+/** Adds a tripId parameter to a URL string. Assumes the URL has no existing parameters. */
+function buildUrlWithParams(baseUrl, tripKey) {
+  return baseUrl + "?" + new URLSearchParams({"tripKey": tripKey}).toString();
```

Comment:
Note that you don't have to put keys in quotes for building an object in JS. Actually, in newer JS there's a feature where, if you put a variable in an object without a key, it assumes you want to assign it to a key with the same name as the variable. So `{"tripKey": tripKey}` is the same as `{tripKey: tripKey}` which is the same a `{tripKey}`.

Note that this convenience setup is why I suggested:
```js
function withParams(baseUrl, params) {
  return baseUrl + "?" + new URLSearchParams(params).toString();
}
```
since you can just write `withParams('/path', {tripKey})`, since it lets you build a URL with any parameters you want including multiple parameters, with any names that you like.

Reply:
ah ok makes sense, fixed it